test/libsolidity/ASTJSONTest.cpp

```
@@ -105,6 +116,15 @@ ASTJSONTest::ASTJSONTest(string const& _filename)
 			);
 			source = string();
 		}
+		else if (boost::algorithm::starts_with(line, failMarker))
+		{
+			string state = line.substr(failMarker.size(), line.size());
+			state.erase(std::remove_if(state.begin(), state.end(),
+									   [](unsigned char x){return std::isspace(x);}),
+						state.end());
+			solAssert(compilerStateMap.count(state), "Unsupported compiler state in sol file");
+			m_expectedFailState = compilerStateMap.at(state);
```

Please report an error if the state is already set (i.e. the there was more than one directive in the file).